sql/2025/05/featured-image-block-usage.sql

```diff
@@ -0,0 +1,77 @@
+# HTTP Archive query to get % WordPress sites using Featured Image in block themes.
+#
+# WPP Research, Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/184
+
+DECLARE
+  DATE_TO_QUERY DATE DEFAULT '2025-04-01';
+
+CREATE TEMPORARY FUNCTION IS_CMS(technologies ARRAY<STRUCT<technology STRING, categories ARRAY<STRING>, info ARRAY<STRING>>>, cms STRING, version STRING) RETURNS BOOL AS (
+  EXISTS(
+    SELECT * FROM UNNEST(technologies) AS technology, UNNEST(technology.info) AS info
+    WHERE technology.technology = cms
+    AND (
+      version = ""
+      OR ENDS_WITH(version, ".x") AND (STARTS_WITH(info, RTRIM(version, "x")) OR info = RTRIM(version, ".x"))
+      OR info = version
+    )
+  )
+);
+
+WITH wordpress AS (
+  SELECT
+    client,
+    page,
+    BOOL(custom_metrics.cms.wordpress.block_theme) AS has_block_theme
+  FROM
+    `httparchive.crawl.pages`
+  WHERE
+    date = DATE_TO_QUERY
+    AND IS_CMS(technologies, 'WordPress', '')
+    AND is_root_page = TRUE
+),
+
+featuredImageBlockDetection AS (
+  SELECT
+    client,
+    page,
+    REGEXP_CONTAINS(response_body, r'<figure class="wp-block-post-featured-image">') AS has_featured_image_block
+  FROM
+    `httparchive.crawl.requests`
+  WHERE
+    date = DATE_TO_QUERY
+    AND is_root_page = TRUE
+    AND is_main_document
+    AND response_body LIKE '%<div class="wp-site-blocks">%'
```

Comment:
I missed before that this was included here. While it is the manual way to detect block theme usage, it is unnecessarily slow compared to the `custom_metrics.cms.wordpress.block_theme` custom metric, which has already stored this information as a simple boolean, in the much faster query for `httparchive.crawl.pages`.

In combination with my proposed change from the previous review, this is also unnecessary to keep as it would limit the results to only include block themes due to the `JOIN`. So from that perspective, we should remove this condition.

To make the query faster though, we can use the condition for featured image block presence here instead, in combination with changing the `JOIN` to a `LEFT JOIN`. This way, this sub query only includes the relevant URLs with featured image block, and by `LEFT JOIN` we'll still keep all results from `httparchive.crawl.pages`, even those without featured image. Those without featured image will then have the value `has_featured_image_block: null`, which in an `IF()` check also evaluates to `false`.
```suggestion
    AND REGEXP_CONTAINS(response_body, r'<figure class="wp-block-post-featured-image">')
```